daemon/task_handler.rs

```diff
@@ -158,6 +159,30 @@ impl TaskHandler {
         state.change_status(task_id, TaskStatus::Running);
     }
 
+    /// As time passes, some delayed tasks may need to be enqueued.
+    /// Gather all stashed tasks and enqueue them if it is after the task's enqueue_at
+    fn check_stashed(&mut self) {
+        let stashed_tasks = {
+            self.state.lock().unwrap().tasks_in_statuses(None, vec![TaskStatus::Stashed]).0
```

Comment:
This function became a little more complicated than it would have needed to.
You can get the mutex on the state once and keep it until everything is finished. This is also important to ensure, that the state won't be manipulated in another thread and thereby gives us a few nice guarantees.

I just did a quick rewrite for you to understand how it may look nicer :)
```
    /// As time passes, some delayed tasks may need to be enqueued.
    /// Gather all stashed tasks and enqueue them if it is after the task's enqueue_at
    fn check_stashed(&mut self) {
        let mut state = self.state.lock().unwrap();
        let stashed_tasks = state.tasks_in_statuses(None, vec![TaskStatus::Stashed]).0;

        for task_id in stashed_tasks {
            // We can unwrap, since we got a mutex on the state and know, that the ids are valid.
            let task = state.tasks.get_mut(&task_id).unwrap();

            if let Some(time) = task.enqueue_at {
                if time <= Local::now() {
                    info!("Delayed task enqueued: {:?}", task);

                    task.status = TaskStatus::Queued;
                    task.enqueue_at = None;
                }
            }
        }
    }
```

Reply:
That is much better, thanks!